fs/metrics/manager/monitors/global_monitor.go

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package monitors
+
+import (
+	"context"
+	"time"
+
+	cm "github.com/awslabs/soci-snapshotter/fs/metrics/common"
+	"github.com/opencontainers/go-digest"
+)
+
+// NewGlobalMonitor returns a new GlobalMonitor. A GlobalMonitor encapsulates the emission of FUSE
+// telemetry data at a global level.
+func NewGlobalMonitor() *GlobalMonitor {
+	return &GlobalMonitor{
+		fuseFailureSignal: make(chan struct{}),
+	}
+}
+
+type GlobalMonitor struct {
+	fuseFailureSignal chan struct{}
+}
+
+// RunFuseFailureListener infinitely listens for any FUSE failure. If one
+// occurs, it increments a metric by 1 and sleeps for a time duration.
+func (gm *GlobalMonitor) RunFuseFailureListener(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-gm.fuseFailureSignal:
+			cm.IncOperationCount(cm.FuseFailureState, digest.Digest(""))
+			time.Sleep(5 * time.Minute)
```

Comment:
nit: since we have a context, we probably want to select over `time.After()` and `ctx.Done()` here instead of just sleeping unconditionally.

Reply:
Good point, will do.